youtube_transcript_api/formatters.py

```diff
@@ -134,13 +134,66 @@ def format_transcripts(self, transcripts, **kwargs):
         """
         return '\n\n\n'.join([self.format_transcript(transcript, **kwargs) for transcript in transcripts])
 
+class SRTFormatter(Formatter):
```

Comment:
It looks like the code here has mostly been copied from `WebVTTFormatter`. Please pull out the shared logic. I also noticed that both use the same `format_transcripts` implementation as `TextFormatter`. Therefore, I think the following class hierarchy would make more sense:
```python
class _TextBasedFormatter(TextFormatter):
    # implement format_transcript and _seconds_to_timestamp here 
    # but move the string format to a method called
    # _format_timestamp(hours, minds, secs, ms)
    # which raises a `NotImplementedError`.

class SRTFormatter(_TextBasedFormatter):
    # implement _format_timestamp

class WebVTTFormatter(_TextBasedFormatter):
    # implement _format_timestamp

Reply:
I just realized that SRT adds an index to each transcript block which WebVTT doesn't. You should be able to work around this in a similar fashion without copying so much code.

Reply:
I updated the class hierarchy in my latest commit. Let me know if there's anything about it you'd like me to change.